src/components/annual-report/section3/useFetchSectionThreeData.js

```diff
@@ -0,0 +1,108 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useState } from 'react'
+
+const section3ByLevel = {
+    data_detail_by_level: {
+        resource: 'analytics.json',
+        params: ({ dataElements, orgUnits, orgUnitLevel, periods }) => ({
+            dimension: `dx:${dataElements.join(';')},ou:${
+                orgUnits.join(';') + ';' + orgUnitLevel
+            },pe:${periods.join(';')}`,
+        }),
+    },
+}
+
+const section3Overall = {
+    data_over_all_org_units: {
+        resource: 'analytics.json',
+        params: ({ dataElements, orgUnits, periods }) => ({
+            dimension: `dx:${dataElements.join(';')},ou:${orgUnits.join(
+                ';'
+            )},pe:${periods.join(';')}`,
+        }),
+    },
+}
+
+export const fetchDataByLevel = async ({ engine, variables }) => {
+    const dataBySubPeriod = await engine.query(section3ByLevel, {
+        variables,
+    })
+    return dataBySubPeriod
+}
+
+export const useFetchSectionThreeData = () => {
+    const [loading, setLoading] = useState(false)
+    const [data, setData] = useState(null)
+    const [error, setError] = useState(null)
+
+    const engine = useDataEngine()
+
+    const refetch = async ({ variables = {} }) => {
+        const { mappedConfiguration, boundaryOrgUnitLevel } = variables
+        const { externalRelations } = mappedConfiguration
+
+        const externalRelationsDEs = [
+            ...externalRelations?.map((rel) => rel.denominator),
+            ...externalRelations?.map((rel) => rel.numerator),
+            ...externalRelations?.map((rel) => rel.externalData),
+        ]
+
+        // set to loading
+        setLoading(true)
+        // get data sets and check that each data element is associated with only one period type
+
+        // we need to preserve information on index to tie response back to relevant denominator relation (some can be filtered out)
+        const byLevelRequestsWithIndices = externalRelations
+            .map((rel, index) => {
+                // if the relation's level is the same, or less the boundary organisation unit level, the byLevel request should be skipped
+                if (!rel.level || rel.level <= boundaryOrgUnitLevel) {
+                    return null
+                }
+
+                return {
+                    request: fetchDataByLevel({
+                        engine,
+                        variables: {
+                            dataElements: [
+                                rel.numerator,
+                                rel.denominator,
+                                rel.externalData,
+                            ],
+                            orgUnits: variables.orgUnits,
+                            orgUnitLevel: `LEVEL-${rel.level}`,
+                            periods: [variables?.currentPeriod.id],
```

Comment:
For me it is over all good to go! 
I only have a question in `useFetchSectionThreeData.js` where the current period is retrieved by id like: `variables?.currentPeriod.id` .
Looking at variables, it looks like current period will be directly assigned a year like :`currentPeriod: '2022',`. 
Does this `.id` refer to the assigned year? 

Also, we can keep the old app's blue color codes for section headers (`#5b92e5`) in case that's a requirement.
Thank you!

Reply:
Thanks, @mauricejulesm !
- for period: I've updated this with the new selector logic on the master branch, but yes, the period.id is the string representation we need to pass to analytics
- I like the blue from the old app, but I was trying to choose one from the standard DHIS2 blues (https://ui.dhis2.nu/principles/color). I think we can decide the actual design here as part of the ticket to format the [tables](https://dhis2.atlassian.net/browse/RWDQA-41).

Reply:
Just a side note, if we go with a blue from the DHIS2 palette, we can add `<CssVariables color />` from the UI library to App.js, then in this CSS file, use `background: var(--colors-blue400);` (I think `blue400`/`#42a5f5` is a good option)